### PR TITLE
Student model

### PR DESCRIPTION
**Pull request by Rauld1**
Creates frisby test file. creates student.json file.  Builds student.spec.js file.

### CODE REVIEW

File: spec/api/students.spec.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+/* jshint quotmark:false */
+
+// Add dependencies
+var frisby = require('frisby');
+// Set up variables for info you will use often
+var url = 'http://localhost:3000/api/students/';
+var initialRecord = {
+  "firstName": "Jim",
+  "lastName": "Smith",
+  "courses": "JavaScript"
```
Comment (jeffreytu) [inline spec/api/students.spec.js]: There's no courses property when you post to api/students/ which is why you get the error below.
<pre><code>Failures:

  1) Frisby Test: Create a contact with post
        [ POST http://localhost:3000/api/students/ ]
   Message:
     Error: Expected undefined 'undefined' to match string 'JavaScript' on key 'courses'
   Stacktrace:
     Error: Expected undefined 'undefined' to match string 'JavaScript' on key 'courses'
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1271:17)
    at jasmine.Matchers.toContainJson (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1186:12)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:717:24)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1074:43)
    at Timer.listOnTimeout [as ontimeout] (timers.js:110:15)

  2) Frisby Test: Enforce mandatory fields when creating
        [ POST http://localhost:3000/api/students/ ]
   Message:
     Error: Expected 'undefined' to be type 'array' for comparison
   Stacktrace:
     Error: Expected 'undefined' to be type 'array' for comparison
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1238:11)
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1250:9)
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1250:9)
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1250:9)
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1250:9)
    at jasmine.Matchers.toContainJson (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1186:12)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:717:24)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1074:43)

  3) Frisby Test: Get contact using id
        [ GET http://localhost:3000/api/students/54c29132024ef9f80e843e01 ]
   Message:
     Error: Expected undefined 'undefined' to match string 'JavaScript' on key 'courses'
   Stacktrace:
     Error: Expected undefined 'undefined' to match string 'JavaScript' on key 'courses'
    at _jsonContains (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1271:17)
    at jasmine.Matchers.toContainJson (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1186:12)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:717:24)
    at null.<anonymous> (C:\Users\Jstadragon\Desktop\starters_academy\fullstack-project-01\node_modules\frisby\lib\frisby.js:1074:43)
    at Timer.listOnTimeout [as ontimeout] (timers.js:110:15)

Finished in 0.135 seconds
4 tests, 15 assertions, 3 failures, 0 skipped</code></pre>

Comment (Rauld1) [inline spec/api/students.spec.js]: Thanks for reviewing it

Comment (Rauld1) [inline spec/api/students.spec.js]: Thanks Jeffrey

**Comment by jeffreytu** [comment]
- In your datasources.json:
```
  "fullstack": {
    "name": "fullstack",
    "connector": "mongodb"
  }
```
Remove this above connection to the local database and put it in datasources.local.json. All of us will be using memory for our datasource. If you want to keep using mongo database locally, putting the above part in local.json will allow you to test with retained information.

- The neukinnis folder needs to be deleted so it doesn't get merged into master.
- This branch is very behind the master, you will need to merge master because the gulpfile doesn't even have a `gulp test:api` for me to run your api spec.

**Comment by goldlilys** [comment]
@Rauld1 Like what jeff said, the gulpfile and neukinnis needs to get the latest from master. To make your life easier, instead of merging master to your branch and have to merge all conflicts, why not clone the new master and create a new branch where you add all the new changes you made, then open a new pull request and close this one? Since the neukinnis directory is gone, you're going to have a lot of merge conflicts to fix things and that might take longer than just moving your new files to a new branch. With starting from current master, most of the fixes for casper and the api that were done during the demo is there, but probably not on your version.

**Comment by goldlilys** [comment]
All tests are passing but your api tests for students.spec.js seems to not  be calling any of the functions? The create action is not inside a function and the postBadRecord is missing.

Other than that, if those missing stuff are fine, this can be merged.

**Pull request merged**